File: AtCoder/ABC/187/python/B_Gentle_Pairs.py
N = int(input())
XY = [list(map(int, input().split())) for _ in range(N)]

res = 0
for i in range(N - 1):
    xi, yi = XY[i]
    for j in range(i + 1, N):
        xj, yj = XY[j]
        if abs(yi - yj) <= abs(xi - xj):
            res += 1
print(res)

File: AtCoder/ABC/187/cpp/B_Gentle_Pairs.cpp
#include <bits/stdc++.h>
#define MAX_N 1000
using namespace std;
typedef long long ll;

ll N;
ll X[MAX_N], Y[MAX_N];

void solve() {
  ll res = 0;
  for (int i = 0; i < N - 1; ++i) {
    for (int j = i + 1; j < N; ++j) {
      if (abs(Y[j] - Y[i]) <= abs(X[j] - X[i])) { ++res; }
    }
  }
  printf("%lld\n", res);
}

int main() {
  cin >> N;
  for (int i = 0; i < N; ++i) {
    cin >> X[i] >> Y[i];
  }
  solve();
}